junifer/markers/reho/tests/test_reho_map_comparison.py

```diff
@@ -0,0 +1,95 @@
+"""Provide tests for ReHo map compute comparison."""
+
+# Authors: Synchon Mandal <s.mandal@fz-juelich.de>
+# License: AGPL
+
+from pathlib import Path
+from typing import Callable, Type
+
+import nibabel as nib
+import numpy as np
+import pytest
+from nibabel.imageclasses import PARRECImage
+
+from junifer.datareader.default import DefaultDataReader
+from junifer.markers.reho.reho_base import ReHoBase
+from junifer.testing.datagrabbers import PartlyCloudyTestingDataGrabber
+
+
+@pytest.fixture
+def reho_map_comparer() -> Type:
+    """Generate class for ReHo map comparison."""
+
+    class ReHoMapComparer(ReHoBase):
+        """Class for comparing ReHo maps."""
+
+        def _load_afni_reho_map(self, subject: str) -> "PARRECImage":
+            """Load afni generated reho map for `subject`.
+
+            Parameters
+            ----------
+            subject : {"sub-01", "sub-02", "sub-03"}
+                The subject to load reho map for.
+
+            Returns
+            -------
+            Niimg-like object
+                The loaded .nii file.
+
+            """
+            return nib.load(
+                Path(__file__).parent / subject / f"{subject}-reho-map.nii"
+            )
+
+        def compute(self, subject: str) -> None:
+            """Compute and compare.
+
+            Parameters
+            ----------
+            subject : {"sub-01", "sub-02", "sub-03"}
+                The subject to load reho map for.
+
+            """
+            # Load development fmri datagrabber
+            dg = PartlyCloudyTestingDataGrabber(
+                reduce_confounds=True, age_group="adult"
+            )
+            # Load datareader
+            dr = DefaultDataReader()
+            with dg:
+                path_info = dg[subject]
+                loaded_data = dr.fit_transform(path_info)
+
+            # Compute python implementation
+            self.use_afni = False
+            python_reho_map = self.compute_reho_map(loaded_data["BOLD"])
+            # Load afni implementation data
+            afni_reho_map = self._load_afni_reho_map(subject)
+            # Compare
+            assert np.testing.assert_allclose(
+                python_reho_map.get_fdata(), afni_reho_map.get_fdata()
+            )
+
+    return ReHoMapComparer
+
+
+@pytest.mark.skip(
+    "should not run due to minor inconsistencies between implementations"
```

Comment:
The comparison might not be correct, but we can still test the implementation separately:

Things to check:
* shape
* values should be normalized

Reply:
I think it's better to leave this module as it is and try to get it tested by having afni in the CI as we had planned.